amp_conf/htdocs/admin/libraries/BMO/ConfigFile.class.php

```diff
@@ -83,25 +83,21 @@ public function removeEntry($section, $key, $val = null) {
 		if (!isset($this->config->ProcessedConfig[$section]))
 			throw new \Exception("Tried to remove key $key from section $section, but that section doesn't exist");
 
-		if (is_array($this->config->ProcessedConfig[$section][$key])) {
-			if ($val == null)
-				throw new \Exception("Sorry, you can't delete an entire section this way, as it's likely a bug");
-
-			// Ok, it's buried in here somewhere. Sigh.
-			foreach ($this->config->ProcessedConfig[$section][$key] as $id => $v) {
-				if ($v == $val) {
-					// Yay, found it.
-					unset($this->config->ProcessedConfig[$section][$key][$id]);
-					// Note, keep going - could be dupes. Don't break here.
-				}
-			}
-
-			// Have we deleted everything from that $key?
-			if (count($this->config->ProcessedConfig[$section][$key]) == 0)
+		if (isset($this->config->ProcessedConfig[$section][$key])) {
+			if (is_array($this->config->ProcessedConfig[$section][$key])) {
```

Comment:
Oh, and not a copy/paste error. Where I was seeing problems was with modules.conf and I'd get tons of undefined index errors for "noload" on each reload. The section was there, but the key wasn't. Section presence is already checked above on line 83, exception is thrown if it isn't there.